lib/lambda_ethereum_consensus/fork_choice/handlers.ex

```diff
@@ -52,7 +52,7 @@ defmodule LambdaEthereumConsensus.ForkChoice.Handlers do
 
     cond do
       # Parent block must be known
-      not Map.has_key?(store.block_states, block.parent_root) ->
+      Store.get_state(store, block.parent_root) |> is_nil() ->
```

Comment:
The state fetched here is later used for state transition, so just storing a boolean wouldn't really help. This PR introduces a ~40% regression, but since we moved everything to the DB, it's to be expected. We can improve it later with caching and pre-fetching.